addons/web/static/tests/core/name_and_signature.test.js

```diff
@@ -81,3 +81,21 @@ test("test name_and_signature widget with noInputName and without name", async f
     expect(".card-header .active").toHaveCount(1);
     expect(".card-header .active").toHaveText("Draw");
 });
+
+test("test name_and_signature widget default signature", async function () {
+    const props = {
+        signature: {
+            name: "Brandon Freeman",
+            signatureImage: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+BCQAHBQICJmhD1AAAAABJRU5ErkJggg==",
+        },
+        mode: "draw",
+        signatureType: "signature",
+        noInputName: true,
+    };
+    await mountWithCleanup(NameAndSignature, { props });
+    const canvas = document.querySelector('canvas');
+    const canvasBuffer = new Uint32Array(canvas.getContext('2d').getImageData(0, 0, canvas.width, canvas.height).data.buffer)
+    const hasSomeDrawing = canvasBuffer.some(color => color !== 0);
```

Comment:
Wouldn't it be enough to check `isSignatureEmpty` ?

Reply:
Indeed, I updated the code. It fails when no signatureImage is provided and it passes with the signatureImage value. Thanks.